src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionCommentFilter.java

```diff
@@ -285,7 +285,7 @@ public boolean accept(TreeWalkerAuditEvent event) {
             // contents and tag suppressions
             final FileContents currentContents = event.getFileContents();
 
-            if (getFileContents() != currentContents) {
+            if (currentContents != null) {
```

Comment:
These 2 are not the same. The old code was checking if the file contents cached matches the incoming file contents, meaning we are trying to process the same file twice. The new code is just checking if we processed a file once, and don't care if it is the same or different.

This is a bad change.

Reply:
ok, we need here pure unit test.

create nested class in SuppressionCommentFilterTest that implments CommentListener.
Make implemention to throw exception on send call of `contents.getSingleLineComments(), contents.getBlockComments()` 
please create test method that create 2 events with such exceptional FileContents instance, pass them into accept method of module. Second "accept" should throw exception.

Reply:
The problem is how do we check file is only processed once in a test.

Reply:
test with mock is provided